codenav/src/goto_file.c

```diff
@@ -188,14 +190,16 @@ directory_check(GtkEntry* entry, GtkEntryCompletion* completion)
     log_debug("New completion list!");
 
     if ( g_path_is_absolute(new_dir) )
-        new_dir_path = new_dir;
+        new_dir_path = g_strdup(new_dir);
     else
         new_dir_path = g_build_filename(directory_ref, new_dir, NULL);
 
     /* Build the new file list for completion */
     completion_list = build_file_list(new_dir_path, new_dir);
   	gtk_entry_completion_set_model (completion, completion_list);
     g_object_unref(completion_list);
+
+    g_free(new_dir_path);
```

Comment:
Shouldn't also `curr_dir` be freed here?

Reply:
`curr_dir` is declared static and freed before it is assigned with a new value at line 182. But of course, since it is static within the function, the last instance will never be freed.
Since I don't know the code and don't want to risk to break more than it helps, it would keep it as is.

Reply:
Oh yeah, I missed the `static`. It's fine then.